src/zap/zapinfo.cpp

```diff
@@ -447,6 +447,25 @@ void ZapInfo::CompileMethod()
         return;
     }
 
+#if defined(_TARGET_X86_) || defined(_TARGET_AMD64_)
+    if (methodAttribs & CORINFO_FLG_JIT_INTRINSIC)
+    {
+        // Skip generating hardware intrinsic method bodies.
```

Comment:
Just to confirm I understand, this means that we will inline the intrinsics where they are called as part of AOT. That is, for:

```cs
Vector128<float> Method()
{
    return Vector128<float>.Zero
}
```

we will directly emit something like the following as part of AOT:
```asm
xorps xmm0, xmm0
ret
```

The exception to this is for the intrinsics themselves, where we will say it can't be AOT'd and will fallback to the JIT, which is to say:
```asm
public struct Vector128<T>
{
    public Vector128<T> Zero
    {
        get
        {
            return Zero;
        }
    }
}
```

We will generate no AOT assembly and will force it to be JIT'd at runtime instead.

Is that correct?

Reply:
Correct. Also, this only applies to CoreLib. We don't AOT compile any of the HW intrinsics elsewhere.